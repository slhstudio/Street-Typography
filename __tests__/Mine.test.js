import React from 'react';
import MineComponent from '../client/components/Mine';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const photos1 = [];
const photos2 = ['photo1', 'photo2', 'photo3']

const fakeProps = {
  handleLogIn: jest.fn()
}

describe('<Mine/>', () => {
  it ('renders and displays properly if there is no user', () => {
    const wrapper = shallow(<MineComponent isUser={false}{...fakeProps}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it ('renders and displays properly if there IS a user', () => {
    const wrapper = shallow(<MineComponent isUser={true}{...fakeProps}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it ('renders and displays properly if there IS a user and no photos yet', () => {
    const wrapper = shallow(<MineComponent isUser={true}{...fakeProps}/>);
    wrapper.setState({ loading : false, photos : photos1 });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it ('renders and displays properly if there IS a user and there are photos', () => {
    const wrapper = shallow(<MineComponent isUser={true}{...fakeProps}/>);
    wrapper.setState({ loading : false, photos : photos2 });
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});