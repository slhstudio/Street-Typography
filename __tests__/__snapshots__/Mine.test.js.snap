// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mine/> renders and displays properly if there IS a user 1`] = `
<div>
  <Loading
    speed={300}
    text="LOADING"
  />
</div>
`;

exports[`<Mine/> renders and displays properly if there IS a user and no photos yet 1`] = `
<div>
  <p>
    You haven't added any photos yet. Get Going!
  </p>
</div>
`;

exports[`<Mine/> renders and displays properly if there IS a user and there are photos 1`] = `
<div>
  <PhotoGrid
    notes={Array []}
    photos={
      Array [
        "photo1",
        "photo2",
        "photo3",
      ]
    }
  />
</div>
`;

exports[`<Mine/> renders and displays properly if there is no user 1`] = `
<div>
  <p>
    You must be logged in to see your photos.
  </p>
</div>
`;
